tests/test_unit_conversion.py

```diff
@@ -20,3 +25,94 @@ def test_add_preferred_custom_unit_name(model):
     # again
     model.units.add_preferred_custom_unit_name('millisecond', [{'prefix': 'milli', 'units': 'second'}])
     assert str(model.units.summarise_units(time_var)) == "millisecond"
+
+
+def test_conversion_factor_original(simple_model):
+    simple_model.get_equation_graph(True)  # set up the graph - it is not automatic
+    symbol_b1 = simple_model.get_symbol_by_cmeta_id("b_1")
+    equation = simple_model.get_equations_for([symbol_b1])
+    factor = simple_model.units.get_conversion_factor(quantity=1 * simple_model.units.summarise_units(equation[0].lhs),
+                                                      to_unit=simple_model.units.ureg('us').units)
+    assert factor == 1000
+
+
+def test_conversion_factor_bad_types(simple_model):
+    simple_model.get_equation_graph(True)  # set up the graph - it is not automatic
+    symbol_b1 = simple_model.get_symbol_by_cmeta_id("b_1")
+    equation = simple_model.get_equations_for([symbol_b1])
+    expression = equation[0].lhs
+    to_unit = simple_model.units.ureg('us').units
+    from_unit = simple_model.units.summarise_units(expression)
+    quantity = 1 * from_unit
+    # no source unit
+    try:
```

Comment:
These won't fail the test if no exception is raised! You should use something like
```python
with pytest.raises(AssertionError) as excinfo:
    simple_model.units.get_conversion_factor(to_unit=to_unit)
assert 'No unit...' in excinfo.value
```
or
```python
with pytest.raises(AssertionError, match='^No unit given.*'):
    simple_model.units.get_conversion_factor(to_unit=to_unit)
```